index.js

```diff
@@ -63,16 +67,17 @@ class Emittery {
 		const staticAnyListeners = [...anyListeners];
 
 		await resolvedPromise;
-		return Promise.all([
-			...staticListeners.map(async listener => {
-				if (listeners.has(listener)) {
-					return listener(eventData);
-				}
+
+		await Promise.all([
+			...staticListeners.map(listener => {
+				return listeners.has(listener) && new Promise(resolve => {
+					resolve(listener(eventData));
+				}).catch(hostReportError);
```

Comment:
IIRC I switched this to `new Promise()`, but @dinoboff do you think we should use `try-await-catch` instead?

Reply:
I prefer `new promise` in case, but I don't mind.

Reply:
I think it should use try-await-catch.